core/src/main/scala/com/spotify/featran/FeatureExtractor.scala

```diff
@@ -90,14 +102,14 @@ class FeatureExtractor[M[_]: CollectionType, T] private[featran] (
    *           `DenseVector[Double]`
    */
   def featureResults[F: FeatureBuilder: ClassTag]: M[FeatureResult[F, T]] = {
-    val fb = CrossingFeatureBuilder(implicitly[FeatureBuilder[F]].newBuilder, fs.crossings)
-    as.cross(aggregate).map {
-      case ((o, a), c) =>
-        fs.featureValues(a, c, fb)
-        FeatureResult(fb.result, fb.rejections, o)
+    val fb = implicitly[FeatureBuilder[F]].newBuilder
+    as.cross(aggregate).cross(fs).map {
```

Comment:
`as.cross((aggregate, fs))` to remove an extra side input operation?

Reply:
@nevillelyh so this is where it gets tricky ... `fs: FeatureSet[T]` is now `fs: M[FeatureSet[T]]`